test/tests.bats

```diff
@@ -6,7 +6,7 @@ load 'test_helper/bats-assert/load'
 
 @test "checking configuration: hostname/domainname" {
   run docker run `docker inspect --format '{{ .Config.Image }}' mail`
-  assert_failure
+  assert_success
```

Comment:
Why did you change this test? It does not seems to be related to your change?

Reply:
Supervisor is now the main process (PID 1) and we can't stop it in the start-mailserver.sh script in a way that would result in a failure. Supervisor is not designed to stop with an error if one of its processes is failing.

So there are basically two options:
1. stopping supervisor gracefully or
2. using a different init process.

I think 1) should be ok, since the container only failed if the setup was failing (due to configuration errors). For correctly configured containers this would not matter.

For option 2) we could either (a) use docker's built-in init process (docker run --init; or docker-compose option "init: true") or (b) add an external init program (like tini) and kill supervisor non-gracefully.

Option 2a) could be problematic for already deployed instances, since the docker-compose files (or docker run commands) need to be changed. I did not check out options for 2b)